```
Error running mint automation
+ mkdir -p 7617-0694818/gopath/src/github.com/minio
+ git -C 7617-0694818/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 7617-0694818/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 7617-0694818/gopath/src/github.com/minio/minio fetch --quiet minio pull/7617/head:pr7617
+ git -C 7617-0694818/gopath/src/github.com/minio/minio checkout --quiet pr7617
+ GO111MODULE=on
+ GOPATH=/mnt/nvme/mint-auto/7617-0694818/gopath
+ make -C 7617-0694818/gopath/src/github.com/minio/minio --quiet
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding github.com/Azure/go-autorest v11.7.0+incompatible
go: finding github.com/sirupsen/logrus v1.4.2
go: finding github.com/gorilla/mux v1.7.0
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/minio/mc v0.0.0-20190529152718-f4bb0b8850cb
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding gopkg.in/Shopify/sarama.v1 v1.20.0
go: finding github.com/DataDog/zstd v1.4.0
go: finding github.com/klauspost/compress v1.5.0
go: finding github.com/minio/sha256-simd v0.1.0
go: finding github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding github.com/nats-io/gnatsd v1.4.1
go: finding contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding golang.org/x/net v0.0.0-20190522155817-f3200d17e092
go: finding github.com/hashicorp/go-version v1.1.0
go: finding github.com/minio/lsync v1.0.1
go: finding github.com/nats-io/nats-server v1.4.1
go: finding github.com/minio/cli v1.21.0
go: finding golang.org/x/sys v0.0.0-20190422165155-953cdadca894
go: finding github.com/konsorten/go-windows-terminal-sequences v1.0.1
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/minio/gokrb5/v7 v7.2.5
go: finding github.com/grpc-ecosystem/grpc-gateway v1.8.5
go: finding github.com/minio/hdfs/v3 v3.0.1
go: finding github.com/klauspost/cpuid v1.2.1
go: finding golang.org/x/sys v0.0.0-20190304154630-e844e0132e93
go: finding go.uber.org/zap v1.10.0
go: finding gopkg.in/ini.v1 v1.42.0
go: finding golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
go: finding github.com/posener/complete v1.2.1
go: finding github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding go.opencensus.io v0.21.0
go: finding github.com/lib/pq v1.0.0
go: finding github.com/jtolds/gls v4.20.0+incompatible
go: finding github.com/rs/cors v1.6.0
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding golang.org/x/sys v0.0.0-20181107165924-66b7b1311ac8
go: finding google.golang.org/api v0.4.0
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/golang/protobuf v1.3.1
go: finding github.com/tidwall/gjson v1.2.1
go: finding go.uber.org/atomic v1.3.2
go: finding github.com/pkg/profile v1.3.0
go: finding golang.org/x/net v0.0.0-20190827160401-ba9fcec4b297
go: finding github.com/segmentio/go-prompt v1.2.1-0.20161017233205-f0d19b6901ad
go: finding golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25
go: finding github.com/hashicorp/raft v1.1.0
go: finding github.com/hashicorp/vault v1.1.0
go: finding github.com/coreos/go-semver v0.3.0
go: finding github.com/mattn/go-isatty v0.0.7
go: finding github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: finding github.com/golang/protobuf v1.2.0
go: finding cloud.google.com/go v0.37.2
go: finding github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: finding github.com/go-sql-driver/mysql v1.4.1
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding google.golang.org/genproto v0.0.0-20180817151627-c66870c02cf8
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/minio/minio v0.0.0-20190510004154-ac3b59645e92
go: finding github.com/klauspost/pgzip v1.2.1
go: finding golang.org/x/crypto v0.0.0-20190820162420-60c769a6c586
go: finding github.com/hashicorp/hcl v1.0.0
go: finding go.etcd.io/bbolt v1.3.3
go: finding github.com/rjeczalik/notify v0.9.2
go: finding github.com/coreos/bbolt v1.3.3
go: finding gopkg.in/urfave/cli.v1 v1.20.0
go: finding github.com/prometheus/common v0.2.0
go: finding golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/tidwall/pretty v1.0.0
go: finding github.com/pborman/getopt v0.0.0-20180729010549-6fdd0a2c7117
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding google.golang.org/grpc v1.19.0
go: finding golang.org/x/sys v0.0.0-20180926160741-c2ed4eda69e7
go: finding github.com/google/go-cmp v0.2.0
go: finding github.com/minio/minio-go/v6 v6.0.29
go: finding honnef.co/go/tools v0.0.0-20190106161140-3f1c8253044a
go: finding github.com/mitchellh/cli v1.0.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/fatih/structs v1.1.0
go: finding github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/valyala/fastjson v1.4.1
go: finding go.opencensus.io v0.19.2
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/aws/aws-sdk-go v1.20.21
go: finding github.com/stretchr/testify v1.2.2
go: finding github.com/go-kit/kit v0.8.0
go: finding honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099
go: finding github.com/stretchr/objx v0.1.1
go: finding github.com/stretchr/testify v1.3.0
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding github.com/smartystreets/goconvey v0.0.0-20190330032615-68dc04aab96a
go: finding github.com/gernest/wow v0.1.0
go: finding github.com/minio/minio-go/v6 v6.0.26
go: finding gopkg.in/h2non/filetype.v1 v1.0.5
go: finding golang.org/x/sys v0.0.0-20180830151530-49385e6e1522
go: finding google.golang.org/api v0.3.0
go: finding github.com/tidwall/sjson v1.0.4
go: finding github.com/golang/protobuf v1.1.0
go: finding github.com/fatih/color v1.7.0
go: finding github.com/posener/complete v1.1.1
go: finding github.com/satori/go.uuid v1.2.0
go: finding github.com/dnaeon/go-vcr v1.0.1
go: finding github.com/kr/logfmt v0.0.0-20140226030751-b84e30acd515
go: finding golang.org/x/lint v0.0.0-20181026193005-c67002cb31c3
go: finding github.com/apache/thrift v0.12.0
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: finding github.com/ncw/directio v1.0.5
go: finding github.com/dustin/go-humanize v1.0.0
go: finding golang.org/x/tools v0.0.0-20190114222345-bf090417da8b
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405
go: finding github.com/pkg/errors v0.8.0
go: finding github.com/golang/mock v1.2.0
go: finding github.com/minio/sio v0.2.0
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/prometheus/client_golang v0.9.1
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding golang.org/x/text v0.3.1-0.20180807135948-17ff2d5776d2
go: finding golang.org/x/net v0.0.0-20181220203305-927f97764cc3
go: finding github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding golang.org/x/text v0.3.0
go: finding github.com/rogpeppe/fastuuid v0.0.0-20150106093220-6724a57986af
go: finding github.com/minio/cli v1.20.0
go: finding golang.org/x/tools v0.0.0-20190312170243-e65039ee4138
go: finding golang.org/x/crypto v0.0.0-20190103213133-ff983b9c42bc
go: finding golang.org/x/net v0.0.0-20190125091013-d26f9f9a57f3
go: finding gopkg.in/yaml.v2 v2.0.0-20170812160011-eb3733d160e7
go: finding github.com/golang/mock v1.1.1
go: finding golang.org/x/lint v0.0.0-20190301231843-5614ed5bae6f
go: finding github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c
go: finding github.com/go-stack/stack v1.8.0
go: finding github.com/baiyubin/aliyun-sts-go-sdk v0.0.0-20180326062324-cfa1a18b161f
go: finding github.com/davecgh/go-spew v1.1.0
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/prometheus/client_model v0.0.0-20190115171406-56726106282f
go: finding golang.org/x/exp v0.0.0-20190121172915-509febef88a4
go: finding github.com/google/martian v2.1.0+incompatible
go: finding github.com/bgentry/speakeasy v0.1.0
go: finding github.com/stretchr/objx v0.1.0
go: finding golang.org/x/time v0.0.0-20181108054448-85acf8d2951c
go: finding github.com/minio/sha256-simd v0.0.0-20190328051042-05b4dd3047e5
go: finding google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: finding golang.org/x/net v0.0.0-20180826012351-8a410e7b638d
go: finding google.golang.org/api v0.2.0
go: finding github.com/client9/misspell v0.3.4
go: finding golang.org/x/sync v0.0.0-20180314180146-1d60e4601c6f
go: finding github.com/hashicorp/go-hclog v0.9.1
go: finding golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
go: finding github.com/alecthomas/participle v0.2.1
go: finding golang.org/x/sys v0.0.0-20190116161447-11f53e031339
go: finding golang.org/x/lint v0.0.0-20190227174305-5b3e6a55c961
go: finding github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d
go: finding gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
go: finding git.apache.org/thrift.git v0.12.0
go: finding cloud.google.com/go v0.26.0
go: finding github.com/nats-io/go-nats-streaming v0.4.4
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/mattn/go-isatty v0.0.3
go: finding github.com/ryanuber/columnize v2.1.0+incompatible
go: finding github.com/nats-io/nats.go v1.8.0
go: finding golang.org/x/sys v0.0.0-20181116152217-5ac8a444bdc5
go: finding github.com/klauspost/reedsolomon v1.9.1
go: finding github.com/mitchellh/go-wordwrap v1.0.0
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/smartystreets/goconvey v0.0.0-20190222223459-a17d461953aa
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding gopkg.in/resty.v1 v1.12.0
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/armon/go-metrics v0.0.0-20190430140413-ec5e00d3c878
go: finding golang.org/x/net v0.0.0-20190301231341-16b79f2e4e95
go: finding github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: finding github.com/mattn/go-colorable v0.0.9
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding gopkg.in/yaml.v2 v2.2.1
go: finding github.com/pkg/xattr v0.4.1
go: finding github.com/gorilla/websocket v1.4.1
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/ugorji/go v1.1.4
go: finding github.com/a8m/mark v0.1.1-0.20170507133748-44f2db618845
go: finding github.com/smartystreets/go-aws-auth v0.0.0-20180515143844-0c1422d1fdb9
go: finding golang.org/x/tools v0.0.0-20190226205152-f727befe758c
go: finding golang.org/x/sys v0.0.0-20190222072716-a9d3bda3a223
go: finding golang.org/x/net v0.0.0-20190404232315-eb5bcb51f2a3
go: finding github.com/golang/groupcache v0.0.0-20190702054246-869f871628b6
go: finding github.com/nats-io/nkeys v0.0.2
go: finding github.com/gogo/protobuf v1.2.1
go: finding github.com/klauspost/readahead v1.3.0
go: finding github.com/census-instrumentation/opencensus-proto v0.2.0
go: finding golang.org/x/sync v0.0.0-20181108010431-42b317875d0f
go: finding github.com/gorilla/rpc v1.2.0
go: finding golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be
go: finding golang.org/x/net v0.0.0-20190311183353-d8887717615a
go: finding github.com/marstr/guid v1.1.0
go: finding golang.org/x/crypto v0.0.0-20190513172903-22d7a77e9e5f
go: finding golang.org/x/tools v0.0.0-20190328211700-ab21143f2384
go: finding github.com/jtolds/gls v4.2.1+incompatible
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding golang.org/x/crypto v0.0.0-20181106171534-e4dc69e5b2fd
go: finding github.com/ghodss/yaml v1.0.0
go: finding github.com/armon/go-radix v0.0.0-20180808171621-7fddfc383310
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding github.com/go-logfmt/logfmt v0.3.0
go: finding golang.org/x/crypto v0.0.0-20181203042331-505ab145d0a9
go: finding google.golang.org/appengine v1.4.0
go: finding github.com/julienschmidt/httprouter v1.2.0
go: finding github.com/prometheus/client_model v0.0.0-20180712105110-5c3871d89910
go: finding github.com/jstemmer/go-junit-report v0.0.0-20190106144839-af01ea7f8024
go: finding cloud.google.com/go v0.34.0
go: finding golang.org/x/net v0.0.0-20181114220301-adae6a3d119a
go: finding github.com/minio/dsync/v2 v2.0.0
go: finding golang.org/x/sys v0.0.0-20190826190057-c7b8b68b1456
go: finding github.com/hashicorp/go-msgpack v0.5.5
go: finding github.com/prometheus/procfs v0.0.0-20181005140218-185b4288413d
go: finding golang.org/x/sys v0.0.0-20180823144017-11551d06cbcc
go: finding github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/hashicorp/golang-lru v0.5.0
go: finding go.opencensus.io v0.19.1
go: finding github.com/openzipkin/zipkin-go v0.1.6
go: finding github.com/kisielk/errcheck v1.1.0
go: finding golang.org/x/text v0.3.2
go: finding github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: finding google.golang.org/appengine v1.1.0
go: finding golang.org/x/sys v0.0.0-20190412213103-97732733099d
go: finding github.com/nats-io/nats-streaming-server v0.14.2
go: finding github.com/mattn/go-colorable v0.1.1
go: finding golang.org/x/sys v0.0.0-20181218192612-074acd46bca6
go: finding golang.org/x/net v0.0.0-20180724234803-3673e40ba225
go: finding golang.org/x/net v0.0.0-20190213061140-3a22650c66bd
go: finding github.com/hashicorp/go-retryablehttp v0.5.3
go: finding google.golang.org/api v0.0.0-20181220000619-583d854617af
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/prometheus/client_golang v0.9.2
go: finding github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: finding github.com/fortytw2/leaktest v1.2.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: finding github.com/gorilla/context v1.1.1
go: finding github.com/hashicorp/go-cleanhttp v0.5.0
go: finding github.com/DataDog/datadog-go v2.2.0+incompatible
go: finding github.com/smartystreets/assertions v0.0.0-20190215210624-980c5ac6f3ac
go: finding golang.org/x/sys v0.0.0-20181021155630-eda9bb28ed51
go: finding github.com/eapache/queue v1.1.0
go: finding golang.org/x/build v0.0.0-20190314133821-5284462c4bec
go: finding github.com/mattn/go-isatty v0.0.5
go: finding golang.org/x/sys v0.0.0-20181122145206-62eef0e2fa9b
go: finding github.com/circonus-labs/circonusllhist v0.1.3
go: finding google.golang.org/appengine v1.3.0
go: finding github.com/gogo/protobuf v1.1.1
go: finding github.com/prometheus/common v0.0.0-20181126121408-4724e9255275
go: finding github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/tv42/httpunix v0.0.0-20150427012821-b75d8614f926
go: finding github.com/gorilla/mux v1.6.2
go: finding github.com/jonboulle/clockwork v0.1.0
go: finding gopkg.in/jcmturner/goidentity.v2 v2.0.0
go: finding github.com/Shopify/toxiproxy v2.1.4+incompatible
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/nats-io/go-nats-streaming v0.4.2
go: finding github.com/howeyc/gopass v0.0.0-20170109162249-bf9dde6d0d2c
go: finding github.com/golang/groupcache v0.0.0-20190129154638-5b532d6fd5ef
go: finding github.com/kisielk/gotool v1.0.0
go: finding github.com/minio/cli v1.3.0
go: finding github.com/tarm/serial v0.0.0-20180830185346-98f6abe2eb07
go: finding go.uber.org/zap v1.9.1
go: finding github.com/gopherjs/gopherjs v0.0.0-20181103185306-d547d1d9531e
go: finding golang.org/x/tools v0.0.0-20180221164845-07fd8470d635
go: finding github.com/klauspost/compress v1.4.1
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding golang.org/x/net v0.0.0-20190424112056-4829fb13d2c6
go: finding golang.org/x/tools v0.0.0-20181219222714-6e267b5cc78e
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding cloud.google.com/go v0.31.0
go: finding golang.org/x/sys v0.0.0-20190429190828-d89cdac9e872
go: finding github.com/soheilhy/cmux v0.1.4
go: finding github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/smartystreets/assertions v0.0.0-20190401211740-f487f9de1cd3
go: finding github.com/sirupsen/logrus v1.3.0
go: finding github.com/minio/blazer v0.0.0-20171126203752-2081f5bf0465
go: finding github.com/jellevandenhooff/dkim v0.0.0-20150330215556-f50fe3d243e1
go: finding github.com/minio/hdfs/v3 v3.0.0
go: finding golang.org/x/crypto v0.0.0-20180904163835-0709b304e793
go: finding github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7
go: finding golang.org/x/crypto v0.0.0-20190128193316-c7b33c32a30b
go: finding github.com/sirupsen/logrus v1.2.0
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/onsi/gomega v1.4.3
go: finding github.com/gorilla/websocket v1.4.0
go: finding golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2
go: finding github.com/minio/lsync v0.0.0-20190207022115-a4e43e3d0887
go: finding golang.org/x/tools v0.0.0-20180917221912-90fa682c2a6e
go: finding github.com/grpc-ecosystem/go-grpc-middleware v1.0.0
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/smartystreets/goconvey v0.0.0-20181108003508-044398e4856c
go: finding google.golang.org/appengine v1.2.0
go: finding github.com/eapache/go-resiliency v1.1.0
go: finding github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568
go: finding github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
go: finding github.com/tmc/grpc-websocket-proxy v0.0.0-20190109142713-0ad062ec5ee5
go: finding github.com/klauspost/cpuid v1.2.0
go: finding golang.org/x/sys v0.0.0-20190124100055-b90733256f2e
go: finding github.com/BurntSushi/toml v0.3.1
go: finding go.uber.org/multierr v1.1.0
go: finding github.com/hashicorp/go-uuid v0.0.0-20180228145832-27454136f036
go: finding github.com/tidwall/pretty v0.0.0-20190325153808-1166b9ac2b65
go: finding gopkg.in/ini.v1 v1.41.0
go: finding github.com/google/go-github v17.0.0+incompatible
go: finding golang.org/x/lint v0.0.0-20181217174547-8f45f776aaf1
go: finding github.com/onsi/ginkgo v1.6.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding golang.org/x/net v0.0.0-20181029044818-c44066c5c816
go: finding github.com/anmitsu/go-shlex v0.0.0-20161002113705-648efa622239
go: finding gopkg.in/fsnotify.v1 v1.4.7
go: finding github.com/minio/sio v0.0.0-20190118043801-035b4ef8c449
go: finding github.com/jmespath/go-jmespath v0.0.0-20180206201540-c2b33e8439af
go: finding gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go: finding gopkg.in/inf.v0 v0.9.1
go: finding go4.org v0.0.0-20180809161055-417644f6feb5
go: finding github.com/DataDog/zstd v1.3.5
go: finding github.com/mwitkow/go-conntrack v0.0.0-20161129095857-cc309e4a2223
go: finding github.com/hashicorp/go-rootcerts v1.0.0
go: finding github.com/nats-io/go-nats v1.7.2
go: finding golang.org/x/net v0.0.0-20181201002055-351d144fa1fc
go: finding gopkg.in/cheggaaa/pb.v1 v1.0.28
go: finding golang.org/x/perf v0.0.0-20180704124530-6e6d33e29852
go: finding gopkg.in/jcmturner/gokrb5.v5 v5.3.0
go: finding grpc.go4.org v0.0.0-20170609214715-11d0a25b4919
go: finding github.com/gogo/protobuf v1.2.0
go: finding google.golang.org/genproto v0.0.0-20181029155118-b69ba1387ce2
go: finding golang.org/x/sys v0.0.0-20180909124046-d0be0721c37e
go: finding github.com/gliderlabs/ssh v0.1.1
go: finding contrib.go.opencensus.io/exporter/ocagent v0.4.10
go: finding github.com/mitchellh/go-homedir v1.0.0
go: finding github.com/hashicorp/go-immutable-radix v1.0.0
go: finding github.com/hpcloud/tail v1.0.0
go: finding github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2
go: finding github.com/minio/minio-go v0.0.0-20190227180923-59af836a7e6d
go: finding golang.org/x/oauth2 v0.0.0-20181203162652-d668ce993890
go: finding github.com/jcmturner/gofork v0.0.0-20180107083740-2aebee971930
go: finding github.com/onsi/ginkgo v1.7.0
go: finding golang.org/x/sync v0.0.0-20190227155943-e225da77a7e6
go: finding github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
go: finding golang.org/x/sys v0.0.0-20190130150945-aca44879d564
go: finding github.com/bradfitz/go-smtpd v0.0.0-20170404230938-deb6d6237625
go: finding golang.org/x/crypto v0.0.0-20180723164146-c126467f60eb
go: finding google.golang.org/grpc v1.19.1
go: finding gopkg.in/jcmturner/rpc.v0 v0.0.2
go: finding github.com/smartystreets/assertions v0.0.0-20190116191733-b6c0e53d7304
go: finding github.com/google/go-querystring v1.0.0
go: finding golang.org/x/net v0.0.0-20190108225652-1e06a53dbb7e
go: finding github.com/minio/dsync v1.0.0
go: finding golang.org/x/sys v0.0.0-20181029174526-d69651ed3497
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding github.com/kr/pty v1.1.3
go: finding github.com/fsnotify/fsnotify v1.4.7
go: finding github.com/Shopify/sarama v1.19.0
go: finding github.com/googleapis/gax-go v2.0.0+incompatible
go: finding github.com/openzipkin/zipkin-go v0.1.3
go: finding github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: finding github.com/kr/pretty v0.1.0
go: finding golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33
go: finding github.com/hashicorp/go-msgpack v0.5.4
go: finding honnef.co/go/tools v0.0.0-20180920025451-e3ad64cb4ed3
go: finding github.com/hashicorp/raft v1.0.1
go: finding github.com/circonus-labs/circonus-gometrics v2.3.1+incompatible
go: finding github.com/grpc-ecosystem/grpc-gateway v1.6.2
go: finding github.com/gopherjs/gopherjs v0.0.0-20190328170749-bb2674552d8f
go: finding github.com/hashicorp/go-uuid v1.0.0
go: finding github.com/nats-io/nats-streaming-server v0.14.1
go: finding go.etcd.io/bbolt v1.3.2
go: finding github.com/pascaldekloe/goe v0.1.0
go: finding golang.org/x/oauth2 v0.0.0-20181017192945-9dcd33a902f4
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding google.golang.org/grpc v1.17.0
go: finding google.golang.org/grpc v1.20.1
go: finding github.com/json-iterator/go v1.1.6
go: finding github.com/kr/text v0.1.0
go: finding golang.org/x/crypto v0.0.0-20181030102418-4d3f4d9ffa16
go: finding github.com/prometheus/procfs v0.0.0-20181204211112-1dc9a6cbc91a
go: finding google.golang.org/api v0.0.0-20181030000543-1d582fd0359e
go: finding github.com/nats-io/nats v1.7.2
go: finding github.com/minio/mc v0.0.0-20190401030144-a1355e50e2e8
go: finding go.opencensus.io v0.18.0
go: finding golang.org/x/net v0.0.0-20180906233101-161cd47e91fd
go: finding github.com/coreos/go-systemd v0.0.0-20181012123002-c6f51f82210d
go: finding golang.org/x/crypto v0.0.0-20190426145343-a29dc8fdc734
go: finding github.com/coreos/bbolt v1.3.2
go: finding github.com/grpc-ecosystem/grpc-gateway v1.5.0
go: finding google.golang.org/genproto v0.0.0-20180831171423-11092d34479b
go: finding google.golang.org/grpc v1.14.0
go: finding git.apache.org/thrift.git v0.0.0-20180902110319-2566ecd5d999
go: finding github.com/openzipkin/zipkin-go v0.1.1
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding google.golang.org/api v0.0.0-20180910000450-7ca32eb868bf
go: finding golang.org/x/tools v0.0.0-20180828015842-6cd1fcedba52
go: finding github.com/kr/pty v1.1.1
go: finding honnef.co/go/tools v0.0.0-20180728063816-88497007e858
go: finding golang.org/x/lint v0.0.0-20190313153728-d0100b6bd8b3
go: finding github.com/prometheus/client_golang v0.8.0
go: finding github.com/prometheus/procfs v0.0.0-20180725123919-05ee40e3a273
go: finding golang.org/x/tools v0.0.0-20190311212946-11955173bddd
go: finding github.com/prometheus/common v0.0.0-20180801064454-c7de2306084e
go: finding google.golang.org/genproto v0.0.0-20181219182458-5a97ab628bfb
go: finding google.golang.org/grpc v1.16.0
go: finding github.com/google/pprof v0.0.0-20181206194817-3ea8567a2e57
go: finding golang.org/x/net v0.0.0-20181106065722-10aee1819953
go: finding golang.org/x/lint v0.0.0-20180702182130-06c8688daad7
go: finding github.com/golang/lint v0.0.0-20180702182130-06c8688daad7
go: finding github.com/minio/minio-go v0.0.0-20190313212832-5d20267d970d
go: finding golang.org/x/crypto v0.0.0-20190325154230-a5d413f7728c
go: finding github.com/ugorji/go/codec v0.0.0-20190320090025-2dc34c0b8780
go: finding golang.org/x/net v0.0.0-20190324223953-e3b2ff56ed87
go: finding github.com/minio/minio v0.0.0-20190325204105-0250f7de678b
go: finding golang.org/x/sys v0.0.0-20190322080309-f49334f85ddc
go: finding github.com/coreos/go-semver v0.2.0
go: finding github.com/gorilla/rpc v0.0.0-20160517062331-bd3317b8f670
go: finding gopkg.in/mattn/go-runewidth.v0 v0.0.4
go: finding gopkg.in/cheggaaa/pb.v2 v2.0.6
go: finding github.com/hashicorp/vault-plugin-auth-centrify v0.0.0-20180816201131-66b0a34a58bf
go: finding github.com/hashicorp/go-cleanhttp v0.0.0-20171218145408-d5fe4b57a186
go: finding github.com/mattn/go-isatty v0.0.4
go: finding github.com/SAP/go-hdb v0.14.0
go: finding github.com/minio/dsync v0.0.0-20190104003057-61c41ffdeea2
go: finding github.com/hashicorp/serf v0.8.2
go: finding github.com/hashicorp/vault-plugin-secrets-alicloud v0.0.0-20190131211812-b0abe36195cb
go: finding github.com/gorilla/handlers v0.0.0-20160410185317-66e6c6f01d8d
go: finding github.com/google/gofuzz v0.0.0-20170612174753-24818f796faf
go: finding gopkg.in/square/go-jose.v2 v2.3.0
go: finding github.com/alecthomas/participle v0.0.0-20190103085315-bf8340a459bd
go: finding github.com/fullsailor/pkcs7 v0.0.0-20180613152042-8306686428a5
go: finding github.com/pkg/profile v1.2.1
go: finding github.com/dimchansky/utfbom v1.1.0
go: finding google.golang.org/api v0.0.0-20180916000451-19ff8768a5c0
go: finding github.com/hashicorp/vault-plugin-auth-azure v0.0.0-20190201222632-0af1d040b5b3
go: finding github.com/hashicorp/vault-plugin-auth-alicloud v0.0.0-20190311155555-98628998247d
go: finding gopkg.in/mattn/go-colorable.v0 v0.1.0
go: finding github.com/Azure/go-autorest v11.5.2+incompatible
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20160609142408-bb955e01b934
go: finding k8s.io/apimachinery v0.0.0-20190313115320-c9defaaddf6f
go: finding github.com/go-ldap/ldap v3.0.2+incompatible
go: finding github.com/hashicorp/vault-plugin-secrets-gcpkms v0.0.0-20190116164938-d6b25b0b4a39
go: finding github.com/boombuler/barcode v1.0.0
go: finding github.com/djherbis/atime v0.0.0-20170215084934-89517e96e10b
go: finding github.com/denisenkom/go-mssqldb v0.0.0-20190315220205-a8ed825ac853
go: finding github.com/nats-io/nats-streaming-server v0.12.2
go: finding github.com/hashicorp/vault-plugin-auth-kubernetes v0.0.0-20190201222209-db96aa4ab438
go: finding github.com/pborman/uuid v1.2.0
go: finding github.com/NYTimes/gziphandler v1.1.1
go: finding github.com/rs/cors v0.0.0-20190116175910-76f58f330d76
go: finding github.com/jeffchao/backoff v0.0.0-20140404060208-9d7fd7aa17f2
go: finding github.com/duosecurity/duo_api_golang v0.0.0-20190308151101-6c680f768e74
go: finding github.com/Azure/azure-sdk-for-go v26.4.0+incompatible
go: finding github.com/eapache/go-resiliency v0.0.0-20160104191539-b86b1ec0dd42
go: finding gopkg.in/gorethink/gorethink.v4 v4.1.0
go: finding github.com/gammazero/deque v0.0.0-20190130191400-2afb3858e9c7
go: finding github.com/ory-am/common v0.4.0
go: finding layeh.com/radius v0.0.0-20190118135028-0f678f039617
go: finding github.com/klauspost/reedsolomon v0.0.0-20190210214925-2b210cf0866d
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20170925032315-6fe16293d6b7
go: finding github.com/centrify/cloud-golang-sdk v0.0.0-20190214225812-119110094d0f
go: finding gopkg.in/Shopify/sarama.v1 v1.10.1
go: finding github.com/gammazero/workerpool v0.0.0-20181230203049-86a96b5d5d92
go: finding github.com/hashicorp/vault-plugin-secrets-azure v0.0.0-20181207232500-0087bdef705a
go: finding github.com/klauspost/crc32 v0.0.0-20161016154125-cb6bfca970f6
go: finding golang.org/x/sys v0.0.0-20190318195719-6c81ef8f67ca
go: finding golang.org/x/tools v0.0.0-20190318200714-bb1270c20edf
go: finding github.com/minio/sio v0.0.0-20180327104954-6a41828a60f0
go: finding github.com/minio/highwayhash v0.0.0-20181220011308-93ed73d64169
go: finding github.com/minio/mc v0.0.0-20190311071728-2e612b23d665
go: finding github.com/mattn/go-colorable v0.0.0-20160210001857-9fdad7c47650
go: finding github.com/go-errors/errors v1.0.1
go: finding github.com/hashicorp/vault-plugin-secrets-ad v0.0.0-20190131222416-4796d9980125
go: finding github.com/hashicorp/go-retryablehttp v0.0.0-20180718195005-e651d75abec6
go: finding github.com/opentracing/opentracing-go v1.0.2
go: finding github.com/sirupsen/logrus v1.0.6
go: finding github.com/google/uuid v1.0.0
go: finding contrib.go.opencensus.io/exporter/ocagent v0.4.7
go: finding github.com/dgrijalva/jwt-go v0.0.0-20180921172315-3af4c746e1c2
go: finding github.com/pkg/xattr v0.0.0-20170808190211-56ed87199eba
go: finding github.com/go-sql-driver/mysql v0.0.0-20181218123637-c45f530f8e7f
go: finding github.com/armon/circbuf v0.0.0-20150827004946-bbbad097214e
go: finding github.com/gorilla/context v0.0.0-20160525203319-aed02d124ae4
go: finding github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce29ed
go: finding github.com/streadway/amqp v0.0.0-20160311215503-2e25825abdbd
go: finding github.com/jefferai/jsonx v1.0.0
go: finding github.com/gorhill/cronexpr v0.0.0-20180427100037-88b0669f7d75
go: finding github.com/coreos/go-oidc v2.0.0+incompatible
go: finding github.com/klauspost/cpuid v0.0.0-20160106104451-349c67577817
go: finding github.com/klauspost/pgzip v0.0.0-20180606150939-90b2c57fba35
go: finding github.com/gocql/gocql v0.0.0-20190301043612-f6df8288f9b4
go: finding gopkg.in/mattn/go-isatty.v0 v0.0.4
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/hashicorp/consul v1.4.3
go: finding github.com/opencontainers/image-spec v1.0.1
go: finding github.com/SermoDigital/jose v0.9.1
go: finding github.com/miekg/dns v1.0.14
go: finding github.com/cheggaaa/pb v0.0.0-20160713104425-73ae1d68fe0b
go: finding github.com/keybase/go-crypto v0.0.0-20190312101036-b475f2ecc1fe
go: finding k8s.io/klog v0.2.0
go: finding github.com/coreos/go-semver v0.0.0-20180108230905-e214231b295a
go: finding golang.org/x/sys v0.0.0-20180828065106-d99a578cf41b
go: finding golang.org/x/sys v0.0.0-20190204203706-41f3e6584952
go: finding github.com/spf13/pflag v1.0.3
go: finding github.com/hashicorp/nomad v0.8.7
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding github.com/hashicorp/vault-plugin-secrets-gcp v0.0.0-20190311200649-621231cb86fe
go: finding github.com/samuel/go-zookeeper v0.0.0-20180130194729-c4fab1ac1bec
go: finding github.com/nats-io/go-nats-streaming v0.0.0-20161216191029-077898146bfb
go: finding github.com/go-ini/ini v1.42.0
go: finding github.com/containerd/continuity v0.0.0-20181203112020-004b46473808
go: finding github.com/Microsoft/go-winio v0.4.12
go: finding github.com/hashicorp/go-plugin v0.0.0-20190220160451-3f118e8ee104
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/dancannon/gorethink v4.0.0+incompatible
go: finding github.com/onsi/ginkgo v1.8.0
go: finding github.com/cenkalti/backoff v2.0.0+incompatible
go: finding github.com/michaelklishin/rabbit-hole v1.5.0
go: finding github.com/go-test/deep v1.0.1
go: finding github.com/pquerna/otp v1.1.0
go: finding gotest.tools v2.2.0+incompatible
go: finding github.com/docker/go-units v0.3.3
go: finding github.com/klauspost/compress v1.3.0
go: finding github.com/mitchellh/hashstructure v1.0.0
go: finding github.com/hashicorp/go-syslog v1.0.0
go: finding github.com/ory/dockertest v3.3.4+incompatible
go: finding github.com/hashicorp/go-hclog v0.8.0
go: finding github.com/pkg/profile v0.0.0-20160315012113-19396ecfda37
go: finding github.com/tidwall/gjson v1.1.4
go: finding golang.org/x/crypto v0.0.0-20190313024323-a1f597ede03a
go: finding github.com/bitly/go-hostpool v0.0.0-20171023180738-a3a6125de932
go: finding github.com/mitchellh/go-testing-interface v0.0.0-20171004221916-a61a99592b77
go: finding github.com/gorilla/mux v0.0.0-20160605233521-9fa818a44c2b
go: finding golang.org/x/net v0.0.0-20190318221613-d196dffd7c2b
go: finding github.com/hashicorp/logutils v1.0.0
go: finding github.com/tidwall/gjson v1.1.2
go: finding github.com/mitchellh/iochan v1.0.0
go: finding github.com/araddon/gou v0.0.0-20190110011759-c797efecbb61
go: finding github.com/Jeffail/gabs v1.1.1
go: finding github.com/oklog/run v1.0.0
go: finding github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190315122603-6f9e54af456e
go: finding github.com/lib/pq v0.0.0-20181016162627-9eb73efc1fcc
go: finding github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142
go: finding github.com/hashicorp/vault-plugin-auth-gcp v0.0.0-20190201215414-7d4c2101e7d0
go: finding github.com/garyburd/redigo v1.6.0
go: finding github.com/opencontainers/go-digest v1.0.0-rc1
go: finding github.com/docker/go-connections v0.4.0
go: finding github.com/briankassouf/jose v0.9.1
go: finding github.com/nats-io/nuid v1.0.0
go: finding github.com/grpc-ecosystem/grpc-gateway v1.7.0
go: finding gopkg.in/VividCortex/ewma.v1 v1.1.1
go: finding github.com/hashicorp/go-hclog v0.0.0-20180709165350-ff2cf002a8dd
go: finding go.opencensus.io v0.18.1-0.20181204023538-aab39bd6a98b
go: finding github.com/nsqio/go-nsq v0.0.0-20181028195256-0527e80f3ba5
go: finding github.com/hashicorp/go-rootcerts v0.0.0-20160503143440-6bb64b370b90
go: finding github.com/go-stomp/stomp v2.0.2+incompatible
go: finding github.com/chrismalek/oktasdk-go v0.0.0-20181212195951-3430665dfaa0
go: finding gopkg.in/ory-am/dockertest.v2 v2.2.3
go: finding golang.org/x/crypto v0.0.0-20180820150726-614d502a4dac
go: finding github.com/skyrings/skyring-common v0.0.0-20160324141443-762fd2bfc12e
go: finding github.com/pquerna/cachecontrol v0.0.0-20180517163645-1555304b9b35
go: finding github.com/hashicorp/go-version v0.0.0-20160214002439-2e7f5ea8e27b
go: finding github.com/minio/cli v0.0.0-20170227073228-b8ae5507c0ce
go: finding github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
go: finding github.com/gotestyourself/gotestyourself v2.2.0+incompatible
go: finding github.com/hashicorp/memberlist v0.1.3
go: finding github.com/ryanuber/columnize v0.0.0-20160712163229-9b3edd62028f
go: finding github.com/onsi/gomega v1.5.0
go: finding gopkg.in/fatih/color.v1 v1.7.0
go: finding github.com/ugorji/go v0.0.0-20180628102755-7d51bbe6161d
go: finding github.com/tidwall/match v1.0.0
go: finding github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869
go: finding github.com/hashicorp/vault v0.0.0-20181121181053-d4367e581fe1
go: finding github.com/coreos/etcd v0.0.0-20180622175804-8f6348a97d48
go: finding google.golang.org/grpc v1.18.0
go: finding github.com/coredns/coredns v0.0.0-20180121192821-d4bf076ccf4e
go: finding github.com/nats-io/go-nats v0.0.0-20161120202126-6b6bf392d34d
go: finding github.com/mitchellh/mapstructure v0.0.0-20160808181253-ca63d7c062ee
go: finding github.com/dustin/go-humanize v0.0.0-20150809201405-1c212aae1d02
go: finding gopkg.in/cheggaaa/pb.v1 v1.0.27
go: finding github.com/gomodule/redigo v0.0.0-20190205135352-43fe51054af5
go: finding github.com/hashicorp/mdns v1.0.0
go: finding k8s.io/api v0.0.0-20190313115550-3c12c96769cc
go: finding github.com/mattbaird/elastigo v0.0.0-20170123220020-2fe47fd29e4b
go: finding github.com/opencontainers/runc v0.1.1
go: finding github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5
go: finding github.com/mitchellh/go-testing-interface v1.0.0
go: finding golang.org/x/crypto v0.0.0-20181029021203-45a5f77698d3
go: finding github.com/hashicorp/go-gcp-common v0.0.0-20180425173946-763e39302965
go: finding github.com/segmentio/go-prompt v0.0.0-20161017233205-f0d19b6901ad
go: finding github.com/eclipse/paho.mqtt.golang v0.0.0-20181129145454-379fd9f99ba5
go: finding github.com/cenkalti/backoff v2.1.1+incompatible
go: finding golang.org/x/sys v0.0.0-20181026203630-95b1ffbd15a5
go: finding google.golang.org/genproto v0.0.0-20180918203901-c3f76f3b92d1
go: finding github.com/fatih/structs v0.0.0-20181010231757-878a968ab225
go: finding github.com/smartystreets/goconvey v0.0.0-20190306220146-200a235640ff
go: finding github.com/mitchellh/copystructure v1.0.0
go: finding github.com/asaskevich/govalidator v0.0.0-20180720115003-f9ffefc3facf
go: finding github.com/nats-io/nats v0.0.0-20160916181735-70b70be17b77
go: finding github.com/hashicorp/go.net v0.0.1
go: finding github.com/patrickmn/go-cache v2.1.0+incompatible
go: finding github.com/hashicorp/vault-plugin-secrets-kv v0.0.0-20190315192709-dccffee64925
go: finding gopkg.in/olivere/elastic.v5 v5.0.31
go: finding github.com/elazarl/go-bindata-assetfs v0.0.0-20151224045452-57eb5e1fc594
go: finding github.com/ugorji/go v1.1.2
go: finding golang.org/x/crypto v0.0.0-20190131182504-b8fe1690c613
go: finding github.com/golang/snappy v0.0.0-20170215233205-553a64147049
go: finding github.com/minio/minio v0.0.0-20190206103305-fd4e15c11641
go: finding github.com/sean-/seed v0.0.0-20170313163322-e2103e2c3529
go: finding github.com/hashicorp/raft v1.0.0
go: finding github.com/hashicorp/go-sockaddr v1.0.0
go: finding github.com/gopherjs/gopherjs v0.0.0-20190309154008-847fc94819f9
go: finding github.com/armon/go-metrics v0.0.0-20180917152333-f0300d1749da
go: finding github.com/hashicorp/yamux v0.0.0-20180604194846-3520598351bb
go: finding gopkg.in/fatih/pool.v2 v2.0.0
go: finding golang.org/x/net v0.0.0-20181023162649-9b4f9f5ad519
go: finding github.com/hashicorp/go-memdb v0.0.0-20190306140544-eea0b16292ad
go: finding github.com/mitchellh/gox v0.4.0
go: finding contrib.go.opencensus.io/exporter/stackdriver v0.0.0-20180919222851-d1e19f5c23e9
go: finding google.golang.org/grpc v1.15.0
go: finding github.com/hashicorp/vault-plugin-auth-jwt v0.0.0-20190314211503-86b44673ce1e
go: finding github.com/coreos/etcd v0.0.0-20180703215944-e4425ee79f2f
go: finding github.com/mitchellh/pointerstructure v0.0.0-20170205204203-f2329fcfa9e2
go: finding github.com/mitchellh/reflectwalk v1.0.0
go: finding golang.org/x/net v0.0.0-20180530234432-1e491301e022
go: finding github.com/golang/protobuf v1.3.0
go: finding github.com/hashicorp/go-msgpack v0.5.3
go: finding golang.org/x/sys v0.0.0-20190129075346-302c3dd5f1cc
go: finding gopkg.in/mgo.v2 v2.0.0-20180705113604-9856a29383ce
go: finding google.golang.org/grpc v1.12.0
go: finding github.com/aws/aws-sdk-go v1.15.31
go: finding go.opencensus.io v0.15.0
go: finding gopkg.in/h2non/filetype.v1 v1.0.3
go: finding github.com/dustin/go-humanize v0.0.0-20170228161531-259d2a102b87
go: finding github.com/pascaldekloe/goe v0.0.0-20180627143212-57f6aae5913c
go: finding google.golang.org/appengine v1.0.0
go: finding google.golang.org/api v0.0.0-20180603000442-8e296ef26005
go: finding github.com/minio/sha256-simd v0.0.0-20190131020904-2d45a736cd16
go: finding golang.org/x/oauth2 v0.0.0-20180603041954-1e0a3fa8ba9a
go: finding google.golang.org/genproto v0.0.0-20180601223552-81158efcc9f2
go: finding cloud.google.com/go v0.23.0
go: finding sigs.k8s.io/yaml v1.1.0
go: finding github.com/go-ini/ini v1.25.4
go: finding github.com/jmespath/go-jmespath v0.0.0-20160202185014-0b12d6b521d8
go: downloading github.com/minio/cli v1.21.0
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading github.com/minio/sha256-simd v0.1.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/valyala/fastjson v1.4.1
go: downloading google.golang.org/api v0.4.0
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/minio/sio v0.2.0
go: extracting github.com/minio/sha256-simd v0.1.0
go: downloading golang.org/x/crypto v0.0.0-20190820162420-60c769a6c586
go: extracting github.com/valyala/fastjson v1.4.1
go: extracting github.com/minio/sio v0.2.0
go: downloading github.com/klauspost/readahead v1.3.0
go: extracting github.com/minio/cli v1.21.0
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/tidwall/sjson v1.0.4
go: extracting github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: extracting github.com/minio/minio-go v0.0.0-20190327203652-5325257a208f
go: downloading github.com/klauspost/reedsolomon v1.9.1
go: downloading github.com/minio/minio-go/v6 v6.0.29
go: extracting github.com/ncw/directio v1.0.5
go: extracting github.com/tidwall/sjson v1.0.4
go: extracting github.com/klauspost/readahead v1.3.0
go: downloading github.com/gorilla/handlers v1.4.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting github.com/minio/minio-go/v6 v6.0.29
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting github.com/gorilla/handlers v1.4.0
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/klauspost/reedsolomon v1.9.1
go: extracting golang.org/x/crypto v0.0.0-20190820162420-60c769a6c586
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/gorilla/rpc v1.2.0
go: extracting github.com/cheggaaa/pb v1.0.28
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/fatih/color v1.7.0
go: extracting github.com/fatih/color v1.7.0
go: extracting github.com/gorilla/rpc v1.2.0
go: extracting github.com/golang/snappy v0.0.1
go: downloading github.com/gorilla/mux v1.7.0
go: downloading github.com/klauspost/cpuid v1.2.1
go: downloading cloud.google.com/go v0.37.2
go: downloading github.com/mattn/go-isatty v0.0.7
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/nats-io/stan.go v0.4.5
go: extracting github.com/mattn/go-isatty v0.0.7
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/minio/hdfs/v3 v3.0.1
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading golang.org/x/sys v0.0.0-20190826190057-c7b8b68b1456
go: extracting github.com/pkg/errors v0.8.1
go: downloading golang.org/x/net v0.0.0-20190827160401-ba9fcec4b297
go: extracting gopkg.in/Shopify/sarama.v1 v1.20.0
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/lib/pq v1.0.0
go: extracting cloud.google.com/go v0.37.2
go: extracting github.com/lib/pq v1.0.0
go: downloading github.com/minio/dsync/v2 v2.0.0
go: downloading github.com/mattn/go-runewidth v0.0.4
go: extracting github.com/minio/dsync/v2 v2.0.0
go: downloading github.com/fatih/structs v1.1.0
go: extracting golang.org/x/net v0.0.0-20190827160401-ba9fcec4b297
go: downloading github.com/mattn/go-colorable v0.1.1
go: extracting github.com/klauspost/cpuid v1.2.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting golang.org/x/sys v0.0.0-20190826190057-c7b8b68b1456
go: extracting github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: extracting github.com/fatih/structs v1.1.0
go: downloading google.golang.org/grpc v1.20.1
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/coredns/coredns v1.4.0
go: extracting github.com/mattn/go-colorable v0.1.1
go: downloading github.com/rs/cors v1.6.0
go: extracting github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: downloading github.com/prometheus/common v0.2.0
go: extracting google.golang.org/grpc v1.20.1
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/rs/cors v1.6.0
go: downloading github.com/tidwall/gjson v1.2.1
go: extracting github.com/miekg/dns v1.1.8
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting github.com/prometheus/common v0.2.0
go: extracting github.com/coredns/coredns v1.4.0
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/tidwall/gjson v1.2.1
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/gogo/protobuf v1.2.1
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/googleapis/gax-go v2.0.0+incompatible
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/googleapis/gax-go v2.0.0+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/davecgh/go-spew v1.1.1
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting google.golang.org/api v0.4.0
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: extracting github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading gopkg.in/ini.v1 v1.42.0
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/gogo/protobuf v1.2.1
go: extracting gopkg.in/ini.v1 v1.42.0
go: downloading github.com/tidwall/pretty v1.0.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: downloading github.com/eapache/queue v1.1.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading github.com/minio/mc v0.0.0-20190529152718-f4bb0b8850cb
go: extracting github.com/tidwall/pretty v1.0.0
go: downloading github.com/pierrec/lz4 v2.0.5+incompatible
go: extracting github.com/eclipse/paho.mqtt.golang v1.1.2-0.20190322152051-20337d8c3947
go: downloading github.com/nats-io/nats.go v1.8.0
go: extracting github.com/eapache/queue v1.1.0
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: extracting github.com/minio/mc v0.0.0-20190529152718-f4bb0b8850cb
go: downloading github.com/Azure/go-autorest v11.7.0+incompatible
go: extracting github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting github.com/nats-io/nats.go v1.8.0
go: downloading go.uber.org/atomic v1.3.2
go: downloading golang.org/x/text v0.3.2
go: extracting github.com/Azure/go-autorest v11.7.0+incompatible
go: extracting github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/nats-io/nkeys v0.0.2
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting go.uber.org/atomic v1.3.2
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/pkg/profile v1.3.0
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/nats-io/nkeys v0.0.2
go: downloading github.com/djherbis/atime v1.0.0
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting github.com/djherbis/atime v1.0.0
go: extracting github.com/pkg/profile v1.3.0
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading go.opencensus.io v0.21.0
go: extracting go.opencensus.io v0.21.0
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/minio/highwayhash v1.0.0
go: extracting github.com/minio/highwayhash v1.0.0
go: downloading github.com/minio/lsync v1.0.1
go: extracting gopkg.in/yaml.v2 v2.2.2
go: extracting github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/eapache/go-xerial-snappy latest
go: extracting github.com/minio/lsync v1.0.1
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/klauspost/compress v1.5.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/tidwall/match v1.0.1
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/golang/protobuf v1.3.1
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/golang/protobuf v1.3.1
go: downloading contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: extracting contrib.go.opencensus.io/exporter/ocagent v0.5.0
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/hashicorp/vault v1.1.0
go: extracting github.com/Azure/azure-sdk-for-go v27.0.0+incompatible
go: extracting google.golang.org/genproto v0.0.0-20190307195333-5fe7a883aa19
go: extracting golang.org/x/text v0.3.2
go: downloading github.com/census-instrumentation/opencensus-proto v0.2.0
go: extracting github.com/census-instrumentation/opencensus-proto v0.2.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: extracting github.com/streadway/amqp v0.0.0-20190402114354-16ed540749f6
go: extracting github.com/grpc-ecosystem/grpc-gateway v1.9.0
go: downloading github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: extracting github.com/minio/parquet-go v0.0.0-20190318185229-9d767baf1679
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/marstr/guid v1.1.0
go: extracting github.com/marstr/guid v1.1.0
go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: downloading git.apache.org/thrift.git v0.12.0
go: extracting git.apache.org/thrift.git v0.12.0
go: downloading github.com/satori/go.uuid v1.2.0
go: extracting github.com/satori/go.uuid v1.2.0
go: extracting github.com/hashicorp/vault v1.1.0
go: extracting github.com/klauspost/compress v1.5.0
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.3
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/go-retryablehttp v0.5.3
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/minio/gokrb5/v7 v7.2.5
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
# github.com/minio/minio/cmd
cmd/admin-handlers.go:1630:15: undefined: errors
make: *** [build] Error 2

```

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -21,6 +21,7 @@ import (
 	"context"
 	"encoding/base64"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
@@ -476,7 +477,6 @@ func newLockEntry(l lockRequesterInfo, resource, server string) *madmin.LockEntr
 }
 
 func topLockEntries(peerLocks []*PeerLocks) madmin.LockEntries {
-	const listCount int = 10
 	entryMap := make(map[string]*madmin.LockEntry)
 	for _, peerLock := range peerLocks {
 		if peerLock == nil {
@@ -497,6 +497,7 @@ func topLockEntries(peerLocks []*PeerLocks) madmin.LockEntries {
 		lockEntries = append(lockEntries, *v)
 	}
 	sort.Sort(lockEntries)
+	const listCount int = 10
 	if len(lockEntries) > listCount {
 		lockEntries = lockEntries[:listCount]
 	}
@@ -654,45 +655,60 @@ func (a adminAPIHandlers) DownloadProfilingHandler(w http.ResponseWriter, r *htt
 	}
 }
 
+type healInitParams struct {
+	bucket, objPrefix     string
+	hs                    madmin.HealOpts
+	clientToken           string
+	forceStart, forceStop bool
+}
+
 // extractHealInitParams - Validates params for heal init API.
-func extractHealInitParams(r *http.Request) (bucket, objPrefix string,
-	hs madmin.HealOpts, clientToken string, forceStart bool, forceStop bool,
-	err APIErrorCode) {
+func extractHealInitParams(vars map[string]string, qParms url.Values, r io.Reader) (hip healInitParams, err APIErrorCode) {
+	hip.bucket = vars[string(mgmtBucket)]
+	hip.objPrefix = vars[string(mgmtPrefix)]
 
-	vars := mux.Vars(r)
-	bucket = vars[string(mgmtBucket)]
-	objPrefix = vars[string(mgmtPrefix)]
-
-	if bucket == "" {
-		if objPrefix != "" {
+	if hip.bucket == "" {
+		if hip.objPrefix != "" {
 			// Bucket is required if object-prefix is given
 			err = ErrHealMissingBucket
 			return
 		}
-	} else if isReservedOrInvalidBucket(bucket, false) {
+	} else if isReservedOrInvalidBucket(hip.bucket, false) {
 		err = ErrInvalidBucketName
 		return
 	}
 
 	// empty prefix is valid.
-	if !IsValidObjectPrefix(objPrefix) {
+	if !IsValidObjectPrefix(hip.objPrefix) {
 		err = ErrInvalidObjectName
 		return
 	}
 
-	qParms := r.URL.Query()
 	if len(qParms[string(mgmtClientToken)]) > 0 {
-		clientToken = qParms[string(mgmtClientToken)][0]
+		hip.clientToken = qParms[string(mgmtClientToken)][0]
 	}
 	if _, ok := qParms[string(mgmtForceStart)]; ok {
-		forceStart = true
+		hip.forceStart = true
 	}
 	if _, ok := qParms[string(mgmtForceStop)]; ok {
-		forceStop = true
+		hip.forceStop = true
 	}
+
+	// Invalid request conditions:
+	//
+	//   Cannot have both forceStart and forceStop in the same
+	//   request; If clientToken is provided, request can only be
+	//   to continue receiving logs, so it cannot be start or
+	//   stop;
+	if (hip.forceStart && hip.forceStop) ||
+		(hip.clientToken != "" && (hip.forceStart || hip.forceStop)) {
+		err = ErrInvalidRequest
+		return
+	}
+
 	// ignore body if clientToken is provided
-	if clientToken == "" {
-		jerr := json.NewDecoder(r.Body).Decode(&hs)
+	if hip.clientToken == "" {
+		jerr := json.NewDecoder(r).Decode(&hip.hs)
 		if jerr != nil {
 			logger.LogIf(context.Background(), jerr)
 			err = ErrRequestBodyParse
@@ -731,7 +747,7 @@ func (a adminAPIHandlers) HealHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	bucket, objPrefix, hs, clientToken, forceStart, forceStop, errCode := extractHealInitParams(r)
+	hip, errCode := extractHealInitParams(mux.Vars(r), r.URL.Query(), r.Body)
 	if errCode != ErrNone {
 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(errCode), r.URL)
 		return
@@ -805,8 +821,8 @@ func (a adminAPIHandlers) HealHandler(w http.ResponseWriter, r *http.Request) {
 	info := objectAPI.StorageInfo(ctx)
 	numDisks := info.Backend.OfflineDisks + info.Backend.OnlineDisks
 
-	healPath := pathJoin(bucket, objPrefix)
-	if clientToken == "" && !forceStart && !forceStop {
+	healPath := pathJoin(hip.bucket, hip.objPrefix)
+	if hip.clientToken == "" && !hip.forceStart && !hip.forceStop {
 		nh, exists := globalAllHealState.getHealSequence(healPath)
 		if exists && !nh.hasEnded() && len(nh.currentStatus.Items) > 0 {
 			b, err := json.Marshal(madmin.HealStartSuccess{
@@ -825,11 +841,11 @@ func (a adminAPIHandlers) HealHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	if clientToken != "" && !forceStart && !forceStop {
+	if hip.clientToken != "" && !hip.forceStart && !hip.forceStop {
 		// Since clientToken is given, fetch heal status from running
 		// heal sequence.
 		respBytes, errCode := globalAllHealState.PopHealStatusJSON(
-			healPath, clientToken)
+			healPath, hip.clientToken)
 		if errCode != ErrNone {
 			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(errCode), r.URL)
 		} else {
@@ -840,14 +856,14 @@ func (a adminAPIHandlers) HealHandler(w http.ResponseWriter, r *http.Request) {
 
 	respCh := make(chan healResp)
 	switch {
-	case forceStop:
+	case hip.forceStop:
 		go func() {
 			respBytes, apiErr := globalAllHealState.stopHealSequence(healPath)
 			hr := healResp{respBytes: respBytes, apiErr: apiErr}
 			respCh <- hr
 		}()
-	case clientToken == "":
-		nh := newHealSequence(bucket, objPrefix, handlers.GetSourceIP(r), numDisks, hs, forceStart)
+	case hip.clientToken == "":
+		nh := newHealSequence(hip.bucket, hip.objPrefix, handlers.GetSourceIP(r), numDisks, hip.hs, hip.forceStart)
 		go func() {
 			respBytes, apiErr, errMsg := globalAllHealState.LaunchNewHealSequence(nh)
 			hr := healResp{respBytes, apiErr, errMsg}
@@ -1609,7 +1625,10 @@ func convertValueType(elem []byte, jsonType gjson.Type) (interface{}, error) {
 	case gjson.False, gjson.True:
 		return strconv.ParseBool(str)
 	case gjson.JSON:
-		return gjson.Parse(str).Value(), nil
+		if gjson.Valid(str) {
+			return gjson.Parse(str).Value(), nil
+		}
+		return nil, errors.New("invalid json")
 	case gjson.String:
 		return str, nil
 	case gjson.Number:

--- a/cmd/admin-handlers_test.go
+++ b/cmd/admin-handlers_test.go
@@ -25,13 +25,16 @@ import (
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"reflect"
 	"strings"
 	"sync"
 	"testing"
+	"time"
 
 	"github.com/gorilla/mux"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/madmin"
+	"github.com/tidwall/gjson"
 )
 
 var (
@@ -647,6 +650,152 @@ func TestToAdminAPIErrCode(t *testing.T) {
 		if actualErr != test.expectedAPIErr {
 			t.Errorf("Test %d: Expected %v but received %v",
 				i+1, test.expectedAPIErr, actualErr)
+		}
+	}
+}
+
+func TestTopLockEntries(t *testing.T) {
+	t1 := UTCNow()
+	t2 := UTCNow().Add(10 * time.Second)
+	peerLocks := []*PeerLocks{
+		{
+			Addr: "1",
+			Locks: map[string][]lockRequesterInfo{
+				"1": {
+					{false, "node2", "ep2", "2", t2, t2, ""},
+					{true, "node1", "ep1", "1", t1, t1, ""},
+				},
+				"2": {
+					{false, "node2", "ep2", "2", t2, t2, ""},
+					{true, "node1", "ep1", "1", t1, t1, ""},
+				},
+			},
+		},
+		{
+			Addr: "2",
+			Locks: map[string][]lockRequesterInfo{
+				"1": {
+					{false, "node2", "ep2", "2", t2, t2, ""},
+					{true, "node1", "ep1", "1", t1, t1, ""},
+				},
+				"2": {
+					{false, "node2", "ep2", "2", t2, t2, ""},
+					{true, "node1", "ep1", "1", t1, t1, ""},
+				},
+			},
+		},
+	}
+	les := topLockEntries(peerLocks)
+	if len(les) != 2 {
+		t.Fatalf("Did not get 2 results")
+	}
+	if les[0].Timestamp.After(les[1].Timestamp) {
+		t.Fatalf("Got wrong sorted value")
+	}
+}
+
+func TestExtractHealInitParams(t *testing.T) {
+	mkParams := func(clientToken string, forceStart, forceStop bool) url.Values {
+		v := url.Values{}
+		if clientToken != "" {
+			v.Add(string(mgmtClientToken), clientToken)
+		}
+		if forceStart {
+			v.Add(string(mgmtForceStart), "")
+		}
+		if forceStop {
+			v.Add(string(mgmtForceStop), "")
+		}
+		return v
+	}
+	qParmsArr := []url.Values{
+		// Invalid cases
+		mkParams("", true, true),
+		mkParams("111", true, true),
+		mkParams("111", true, false),
+		mkParams("111", false, true),
+		// Valid cases follow
+		mkParams("", true, false),
+		mkParams("", false, true),
+		mkParams("", false, false),
+		mkParams("111", false, false),
+	}
+	varsArr := []map[string]string{
+		// Invalid cases
+		{string(mgmtPrefix): "objprefix"},
+		// Valid cases
+		{},
+		{string(mgmtBucket): "bucket"},
+		{string(mgmtBucket): "bucket", string(mgmtPrefix): "objprefix"},
+	}
+
+	// Body is always valid - we do not test JSON decoding.
+	body := `{"recursive": false, "dryRun": true, "remove": false, "scanMode": 0}`
+
+	// Test all combinations!
+	for pIdx, parms := range qParmsArr {
+		for vIdx, vars := range varsArr {
+			_, err := extractHealInitParams(vars, parms, bytes.NewBuffer([]byte(body)))
+			isErrCase := false
+			if pIdx < 4 || vIdx < 1 {
+				isErrCase = true
+			}
+
+			if err != ErrNone && !isErrCase {
+				t.Errorf("Got unexpected error: %v %v %v", pIdx, vIdx, err)
+			} else if err == ErrNone && isErrCase {
+				t.Errorf("Got no error but expected one: %v %v", pIdx, vIdx)
+			}
+		}
+	}
+
+}
+
+func TestNormalizeJSONKey(t *testing.T) {
+	cases := []struct {
+		input         string
+		correctResult string
+	}{
+		{"notify.webhook.1", "notify.webhook.:1"},
+		{"notify.webhook.ok", "notify.webhook.ok"},
+		{"notify.webhook.1.2", "notify.webhook.:1.:2"},
+		{"abc", "abc"},
+	}
+	for i, tcase := range cases {
+		res := normalizeJSONKey(tcase.input)
+		if res != tcase.correctResult {
+			t.Errorf("Case %d: failed to match", i)
+		}
+	}
+}
+
+func TestConvertValueType(t *testing.T) {
+	cases := []struct {
+		elem          []byte
+		jt            gjson.Type
+		correctResult interface{}
+		isError       bool
+	}{
+		{[]byte(""), gjson.Null, nil, false},
+		{[]byte("t"), gjson.False, true, false},
+		{[]byte("f"), gjson.False, false, false},
+		{[]byte(`{"a": 1}`), gjson.JSON, map[string]interface{}{"a": float64(1)}, false},
+		{[]byte(`abc`), gjson.String, "abc", false},
+		{[]byte(`1`), gjson.Number, float64(1), false},
+		{[]byte(`a1`), gjson.Number, nil, true},
+		{[]byte(`{"A": `), gjson.JSON, nil, true},
+		{[]byte(`2`), gjson.False, nil, true},
+	}
+	for i, tc := range cases {
+		res, err := convertValueType(tc.elem, tc.jt)
+		if err != nil {
+			if !tc.isError {
+				t.Errorf("Case %d: got an error when none was expected", i)
+			}
+		} else if err == nil && tc.isError {
+			t.Errorf("Case %d: got no error though we expected one", i)
+		} else if !reflect.DeepEqual(res, tc.correctResult) {
+			t.Errorf("Case %d: result mismatch - got %#v", i, res)
 		}
 	}
 }

